{"name": "test_check_first_box_and_navigate", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nNoSuchElementError: An element could not be located on the page using the given search parameters.\n    at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\a.jamshidi\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\n    at AndroidUiautomator2Driver.findElement (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)", "trace": "login_and_dashboard = <pages.dashboard.dashboard_page.DashboardPage object at 0x00000197BD904970>\n\n    @pytest.mark.order(3)\n    @allure.feature(\"Box Page\")\n    @allure.story(\"Check First Box and Navigate to Profile\")\n    @allure.severity(allure.severity_level.NORMAL)\n    def test_check_first_box_and_navigate(login_and_dashboard):\n        \"\"\"تست بررسی نام و مقدار اولین باکس و هدایت به صفحه پروفایل باکس\"\"\"\n    \n        box_page = login_and_dashboard\n        driver = box_page.driver\n    \n        try:\n            with allure.step(\"Click on box icon to open box page\"):\n                logger.info(\"کلیک روی آیکون باکس برای باز کردن صفحه باکس...\")\n                box_page = login_and_dashboard.click_box_icon()\n    \n            with allure.step(\"Close the onboarding page\"):\n                logger.info(\"بستن صفحه انبوردینگ...\")\n                box_page.close_onboarding()\n    \n            with allure.step(\"Check if first box name is correct\"):\n                try:\n                    assert box_page.is_first_box_name_correct(), \"نام اولین باکس نادرست است.\"\n                    logger.info(\"نام اولین باکس صحیح است.\")\n                except AssertionError as e:\n                    logger.warning(f\"نام اولین باکس نادرست است: {e}\")\n    \n            with allure.step(\"Check if first box amount is correct\"):\n                assert box_page.is_first_box_amount_correct(), \"مقدار اولین باکس نادرست است.\"\n                logger.info(\"مقدار اولین باکس صحیح است.\")\n    \n            with allure.step(\"Click on the first box and navigate to profile\"):\n                logger.info(\"کلیک روی اولین باکس و هدایت به صفحه پروفایل باکس...\")\n                box_profile_page = box_page.click_first_box()\n    \n                # تایید نمایش صفحه پروفایل\n                assert box_profile_page.is_page_displayed_deposit_btn(), \"صفحه پروفایل باکس نمایش داده نشده است.\"\n                logger.info(\"صفحه پروفایل باکس نمایش داده شد.\")\n    \n        except Exception as e:\n            logger.error(f\"خطا رخ داد: {e}\")\n            capture_screenshot(driver, \"first_box_test_error\")\n>           raise e\n\ntests\\box\\test_first_box_profile.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ntests\\box\\test_first_box_profile.py:29: in test_check_first_box_and_navigate\n    assert box_page.is_first_box_name_correct(), \"نام اولین باکس نادرست است.\"\npages\\dashboard\\box\\box_page.py:88: in is_first_box_name_correct\n    return self.driver.find_element(*self.first_box_name).get_attribute(\"text\") == self.text_reference[\"box_page\"][\"first_box_name\"]\nC:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\appium\\webdriver\\webdriver.py:382: in find_element\n    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']\nC:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:354: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <appium.webdriver.errorhandler.MobileErrorHandler object at 0x00000197BD904F70>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"An element could not be located on the page u...ode_modules\\\\\\\\appium\\\\\\\\node_modules\\\\\\\\@appium\\\\\\\\base-driver\\\\\\\\lib\\\\\\\\basedriver\\\\\\\\commands\\\\\\\\find.ts:75:12)\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"\n        https://www.w3.org/TR/webdriver/#errors\n        \"\"\"\n        payload = response.get('value', '')\n        if isinstance(payload, dict):\n            payload_dict = payload\n        else:\n            try:\n                payload_dict = json.loads(payload)\n            except (json.JSONDecodeError, TypeError):\n                return\n            if not isinstance(payload_dict, dict):\n                return\n        value = payload_dict.get('value')\n        if not isinstance(value, dict):\n            return\n        error = value.get('error')\n        if not error:\n            return\n    \n        message = value.get('message', error)\n        stacktrace = value.get('stacktrace', '')\n        # In theory, we should also be checking HTTP status codes.\n        # Java client, for example, prints a warning if the actual `error`\n        # value does not match to the response's HTTP status code.\n        exception_class: Type[sel_exceptions.WebDriverException] = ERROR_TO_EXC_MAPPING.get(\n            error, sel_exceptions.WebDriverException\n        )\n        if exception_class is sel_exceptions.WebDriverException and message:\n            if message == 'No such context found.':\n                exception_class = appium_exceptions.NoSuchContextException\n            elif message == 'That command could not be executed in the current context.':\n                exception_class = appium_exceptions.InvalidSwitchToTargetException\n    \n        if exception_class is sel_exceptions.UnexpectedAlertPresentException:\n            raise sel_exceptions.UnexpectedAlertPresentException(\n                msg=message,\n                stacktrace=format_stacktrace(stacktrace),\n                alert_text=value.get('data'),\n            )\n>       raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))\nE       selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       NoSuchElementError: An element could not be located on the page using the given search parameters.\nE           at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\a.jamshidi\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\nE           at processTicksAndRejections (node:internal/process/task_queues:95:5)\nE           at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\nE           at AndroidUiautomator2Driver.findElement (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)\n\nC:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\appium\\webdriver\\errorhandler.py:125: NoSuchElementException"}, "description": "تست بررسی نام و مقدار اولین باکس و هدایت به صفحه پروفایل باکس", "steps": [{"name": "Click on box icon to open box page", "status": "passed", "start": 1728998175499, "stop": 1728998175586}, {"name": "Close the onboarding page", "status": "passed", "start": 1728998175587, "stop": 1728998186881}, {"name": "Check if first box name is correct", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: An element could not be located on the page using the given search parameters.; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nNoSuchElementError: An element could not be located on the page using the given search parameters.\n    at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\a.jamshidi\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\find.ts:86:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:60:12)\n    at AndroidUiautomator2Driver.findElement (C:\\Users\\a.jamshidi\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\basedriver\\commands\\find.ts:75:12)\n", "trace": "  File \"G:\\android_automation\\tests\\box\\test_first_box_profile.py\", line 29, in test_check_first_box_and_navigate\n    assert box_page.is_first_box_name_correct(), \"نام اولین باکس نادرست است.\"\n  File \"G:\\android_automation\\pages\\dashboard\\box\\box_page.py\", line 88, in is_first_box_name_correct\n    return self.driver.find_element(*self.first_box_name).get_attribute(\"text\") == self.text_reference[\"box_page\"][\"first_box_name\"]\n  File \"C:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\appium\\webdriver\\webdriver.py\", line 382, in find_element\n    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']\n  File \"C:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 354, in execute\n    self.error_handler.check_response(response)\n  File \"C:\\Users\\a.jamshidi\\android_automation\\android_automation\\.venv\\lib\\site-packages\\appium\\webdriver\\errorhandler.py\", line 125, in check_response\n    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))\n"}, "start": 1728998186881, "stop": 1728998186964}], "attachments": [{"name": "first_box_test_error_20241015_164626.png", "source": "2afa8abf-0a5f-47fe-9e44-1f6bb61f5914-attachment.png", "type": "image/png"}, {"name": "log", "source": "6edd15e3-29a3-45be-8b3f-afe49ceaa8df-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "4b66138c-3921-4868-af1a-7e19be998ac0-attachment.txt", "type": "text/plain"}], "start": 1728998175499, "stop": 1728998187317, "uuid": "38468e80-301c-4a34-8712-d575e126ca06", "historyId": "4830fe53d7efbeb6d1ea27b285ec02a4", "testCaseId": "4830fe53d7efbeb6d1ea27b285ec02a4", "fullName": "tests.box.test_first_box_profile#test_check_first_box_and_navigate", "labels": [{"name": "severity", "value": "normal"}, {"name": "story", "value": "Check First Box and Navigate to Profile"}, {"name": "feature", "value": "Box Page"}, {"name": "tag", "value": "order(3)"}, {"name": "parentSuite", "value": "tests.box"}, {"name": "suite", "value": "test_first_box_profile"}, {"name": "host", "value": "ajamshidi-lap-qa"}, {"name": "thread", "value": "81652-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.box.test_first_box_profile"}]}